### PR TITLE
added route and changed schema for subscriber

### PR DESCRIPTION
**Pull request by vlokesh08**
Created model for Subscriber and added the routes for the following for the issue #100

### CODE REVIEW

File: backend/src/routes/subscriber.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    return c.json({
+      error: "Credentials failed",
+    });
+  }
+});
+
+subscriberRouter.post("/subscribe", async (c) => {
+  const prisma = new PrismaClient({
+    datasourceUrl: c.env.DATABASE_URL,
+  }).$extends(withAccelerate());
```
Comment (aadeshkulkarni) [inline backend/src/routes/subscriber.ts:39]: This 
```bash
const prisma = new PrismaClient({
    datasourceUrl: c.env.DATABASE_URL,
  })  
```
can now be re-written as 

```bash
const prisma = getDBInstance(c);
```

Just make sure you 
```bash
import { getDBInstance } from "../db/util";
```

Note: Please change it for all routes in subscriber.ts where db interaction is performed.

File: backend/src/routes/subscriber.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  const { userId, subscriberId } = await c.req.json();
+  
+  try {
+    await prisma.subscriber.create({
+      data: {
+        userId,
+        subscriberId,
+      },
+    });
```
Comment (aadeshkulkarni) [inline backend/src/routes/subscriber.ts:49]: There needs to be a validation check in place.

What if the Aadesh has already subscribed to Lokesh ? A duplicate entry would be created if we don't check first.

File: backend/src/routes/subscriber.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    return c.json({
+      message: "Subscribed successfully",
```
Comment (aadeshkulkarni) [inline backend/src/routes/subscriber.ts:51]: We should also return the updated count of subscribers in the response.

File: backend/src/routes/subscriber.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+  try {
+    await prisma.subscriber.deleteMany({
+      where: {
+        userId,
+        subscriberId,
+      },
+    });
+    return c.json({
+      message: "Unsubscribed successfully",
```
Comment (aadeshkulkarni) [inline backend/src/routes/subscriber.ts:76]: We should also return the updated count of subscribers in the response.

**Comment by vlokesh08** [comment]
Made the changes @aadeshkulkarni please review.

**Comment by aadeshkulkarni** [comment]
@vlokesh08 - Please add a short (1min) screen recording of how the endpoints work on Postman.

**Comment by vlokesh08** [comment]
Hi @aadeshkulkarni hereby attaching video reference for the 3 routes

https://github.com/aadeshkulkarni/medium-app/assets/76467286/e47edc04-265b-4575-b423-dfb4d39a38bc

**Pull request merged**